config/package-is-installable/src/checkPlatform.ts

```diff
@@ -45,25 +53,44 @@ export interface Platform {
 
 export type WantedPlatform = Partial<Platform>
 
-function checkList (value: string, list: string | string[]) {
-  let tmp; let match = false; let blc = 0
+function checkList (value: string | string[], list: string | string[]): boolean {
+  let tmp
+  let match = false
+  let blc = 0
+
   if (typeof list === 'string') {
     list = [list]
   }
+
   if (list.length === 1 && list[0] === 'any') {
     return true
   }
-  for (let i = 0; i < list.length; ++i) {
-    tmp = list[i]
-    if (tmp[0] === '!') {
-      tmp = tmp.slice(1)
-      if (tmp === value) {
-        return false
+
+  const values = Array.isArray(value) ? value : [value]
+
+  for (const val of values) {
+    for (let i = 0; i < list.length; ++i) {
+      tmp = list[i]
+      if (tmp[0] === '!') {
+        tmp = tmp.slice(1)
+        if (tmp === val) {
+          return false
+        }
+        ++blc
+      } else {
+        match = match || tmp === val
       }
-      ++blc
-    } else {
-      match = match || tmp === value
     }
   }
   return match || blc === list.length
 }
+
+function dedupeArchs (current: string, supported: string[]) {
+  const result = supported.filter((arch) => arch !== 'current')
+
+  if (supported.includes('current')) {
+    result.push(current)
+  }
+
+  return result
```

Comment:
I guess a more effective way would be to do something like this
```suggestion
  return supportedList.map((supported) => supported === 'current' ? current : supported)
```